extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DataSourcesBuildTimeConfig.java

```diff
@@ -68,13 +69,8 @@ public DataSourceBuildTimeConfig getDataSourceRuntimeConfig(String dataSourceNam
         if (DataSourceUtil.isDefault(dataSourceName)) {
             return defaultDataSource;
         }
-
         DataSourceBuildTimeConfig dataSourceBuildTimeConfig = namedDataSources.get(dataSourceName);
-        if (dataSourceBuildTimeConfig == null) {
-            return new DataSourceBuildTimeConfig();
-        }
-
-        return dataSourceBuildTimeConfig;
+        return Objects.requireNonNullElseGet(dataSourceBuildTimeConfig, DataSourceBuildTimeConfig::new);
```

Comment:
Can't say I find this more readable but I suppose it's a matter of taste.

Reply:
I like the compact form, agree that the naming is not ideal though